backend/src/bundles/user-details/user-details.repository.ts

```diff
@@ -0,0 +1,56 @@
+import { type Repository } from '~/common/types/types.js';
+
+import { type UserDetailsUpdateDto } from './types/types.js';
+import { UserDetailsEntity } from './user-details.entity.js';
+import { type UserDetailsModel } from './user-details.model.js';
+
+class UserDetailsRepository implements Repository {
+    private userDetailsModel: typeof UserDetailsModel;
+
+    public constructor(userDetailsModel: typeof UserDetailsModel) {
+        this.userDetailsModel = userDetailsModel;
+    }
+
+    public async find(
+        payload: Record<string, unknown>,
+    ): Promise<UserDetailsEntity | undefined> {
```

Comment:
instead of undefined return null please
```suggestion
    ): Promise<UserDetailsEntity | null> {
```